src/main/java/com/nincraft/ninbot/components/ac/turnips/generator/TurnipPriceGenerator.java

```diff
@@ -0,0 +1,64 @@
+package com.nincraft.ninbot.components.ac.turnips.generator;
+
+import com.nincraft.ninbot.components.ac.turnips.TurnipPattern;
+import com.nincraft.ninbot.components.ac.turnips.TurnipPricesRepository;
+import com.nincraft.ninbot.components.ac.turnips.TurnipPricesService;
+import lombok.val;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+import static com.nincraft.ninbot.components.ac.turnips.TurnipPattern.*;
+
+@Component
+public class TurnipPriceGenerator {
+
+    private Random random;
+    private TurnipPricesService turnipPricesService;
+
+    public TurnipPriceGenerator(TurnipPricesService turnipPricesService) {
+        this.random = new Random();
```

Comment:
Does this actually need to be set in the constructor? It's only set in the below methods, never immediately accessed.